RZCardEntry/RZCardEntry/CardType.swift

```diff
@@ -0,0 +1,203 @@
+//
+//  CardType.swift
+//  CardEntry
+//
+//  Created by Jason Clark on 6/28/16.
+//  Copyright © 2016 Raizlabs. All rights reserved.
+//
+
+import Foundation
+
+//MARK: - CardType
+
+enum CardType {
+    case visa
+    case masterCard
+    case amex
+    case diners
+    case discover
+    case jcb
+
+    static let allValues: [CardType] = [.visa, .masterCard, .amex, .diners, .discover, .jcb]
+
+    fileprivate var validationRequirements: ValidationRequirement {
+        var prefix = [PrefixContainable](), length = [Int]()
+
+        switch self {
+        /* // IIN prefixes and length requriements retreived from https://en.wikipedia.org/wiki/Bank_card_number on June 28, 2016 */
+
+        case .visa:         prefix = ["4"]
+                            length = [13, 16, 19]
+
+        case .masterCard:   prefix = ["51"..."55", "2221"..."2720"]
+                            length = [16]
+
+        case .amex:         prefix = ["34", "37"]
+                            length = [15]
+
+        case .discover:     prefix = ["6011", "65"]
+                            length = [16]
+
+        case .diners:       prefix = ["300"..."305", "309", "38"..."39"]
```

Comment:
I am in favor of `ClosedInterval`. People treat credit card numbers as strings, whether explicitly or implicitly. When you read one aloud, do you chunk it into space-delimited sequences? Do you talk about the "last four?" Does it "start with" a certain character. Then it's a String.